backend/test/unit/TrialStateChangeEvents.unit.test.js

```diff
@@ -0,0 +1,125 @@
+const Client = require('socket.io-client');
+const MafiaGameMock = require('../mocks/MafiaGameMock');
+const config = require('../../config.json');
+const Player = require('../../domain/Player');
+const roles = require('../../domain/Enum/Role');
+
+describe('trial-start unit tests', () => {
+    const port = process.env.PORT || config.local_port;
+    const roomElements = MafiaGameMock.createMafiaGameWithOnePlayerMock(port);
+
+    // Create a new client, and connect it to the server via a socket
+    let clientSocket;
+    beforeEach((done) => {
+        clientSocket = new Client(`http://localhost:${port}`);
+        clientSocket.on('connect', done);
+    });
+
+    // Disconnect each socket connected to the server
+    afterEach((done) => {
+        const { sockets } = roomElements.io.sockets;
+        sockets.forEach((socket) => {
+            socket.disconnect(true);
+        });
+
+        MafiaGameMock.resetRoom(roomElements.roomID);
+
+        done();
+    });
+
+    // Close the server once all tests are done
+    afterAll(() => {
+        roomElements.socketIOServer.close();
+    });
+
+    test('trial-start successful call, no votes, winning condition is not fulfilled', (done) => {
+        const playerA = new Player(null, null, 'a', roles.MAFIA, true);
+        const playerB = new Player(null, null, 'b', roles.JESTER, true);
+        const playerC = new Player(null, null, 'c', roles.CIVILIAN, true);
+
+        MafiaGameMock.addPlayer(playerA, roomElements.roomID);
+        MafiaGameMock.addPlayer(playerB, roomElements.roomID);
+        MafiaGameMock.addPlayer(playerC, roomElements.roomID);
+
+        clientSocket.on('trial-start', (trialStartDTO) => {
+            expect(trialStartDTO.timeToVote).toBe(config.trial_total_vote_time_in_milliseconds);
+        });
+
+        clientSocket.on('trial-end', (trialEndDTO) => {
+            expect(trialEndDTO.playerKilled).toBe('abstain Vote');
+            expect(trialEndDTO.isGameOver).toBe(false);
+            done();
+        });
+
+        clientSocket.emit('start-trial');
+    });
+
+    test('trial-start successful call, one abstain vote, winning condition is fulfilled', (done) => {
```

Comment:
Just for clarification, what situation is this test trying to test?
Only one player exists in the gamemock, whos the Mafia, and he is set to be dead. 
Then the dead player (statewise) chucks a abstain vote during a trial?.. Sorry just a little lost

Reply:
Hmm, yeap you're absolutely right :) I was blindly trying to test multiple pathways (in this case, testing whether abstain votes work and then we correctly emit that the game is over), but this pathway is pretty unrealistic, as you outlined. Deleted this test!